### ISSUE TITLE
In XAML Islands app with WinUI 2 secondary windows do not re-scale when Windows DPI settings are changed

### ISSUE THREAD

**Issue by lyahdav** [microsoft/microsoft-ui-xaml#6624]
### Describe the bug

We have a XAML Islands app (OS XAML embedded in Win32) which we're adding WinUI 2 to. We copied the following initialization code from another app that uses XAML Islands and WinUI 2: https://github.com/microsoft/react-native-windows/blob/d6d0c7d9c5edb86cf80f236e261ae30e54f4aa81/packages/playground/windows/playground-win32/Playground-Win32.cpp#L390-L396. Everything works fine with WinUI 2 except that when you change your display scale (DPI) in Windows settings, it re-scales the main app window (first window that was created, unrelated to which window is focused). All other windows do not get re-scaled. Secondary windows continue rendering as if the scale wasn't changed until the app is restarted. We are using Per-Monitor DPI Awareness. We listen for the win32 WM_DPICHANGED message and resize all windows appropriately when we get it.

### Steps to reproduce the bug

1. Add WinUI 2 to a XAML Islands (Win32) app
2. Add some code to open multiple Win32 windows in the app
3. Run app and open multiple windows
4. In Windows settings change display scale

### Expected behavior

Expected: all windows re-scale
Actual: only main window re-scales. Restarting app fixes this (but we want to avoid requiring this).

### NuGet package version

Microsoft.UI.Xaml 2.7.0

### Windows app type

- [ ] UWP
- [X] Win32

### Device form factor

Desktop

### Windows version

May 2021 Update (19043)

### Additional context

Note we're using WinUI 2.7.1-prerelease.211026002. That option isn't available in the dropdown above for NuGet package version.

**Comment by lyahdav** [comment]
I tried to reproduce this on https://github.com/microsoft/terminal which also uses XAML Islands and WinUI 2 and I couldn't but I realized there's one key difference: Terminal creates a separate process for each window while in our app it's a single process with a single core window and a single windowing thread.

**Comment by sylveon** [comment]
CNR in my xaml islands app, which uses one thread per window.

**Comment by lyahdav** [comment]
I realized I was incorrect in my OP saying we were using Per-Monitor V2 DPI Awareness (just updated it). We set our DPI Awareness as:
```
<dpiAwareness>PerMonitor, PerMonitorV2</dpiAwareness>
```

This means it will always prefer `PerMonitor`. When I change it to only use `PerMonitorV2`, it fixes this issue. Given the minimum Windows version for XAML Islands is 1903 and `PerMonitorV2` was introduced in 1703, there's no need to have `PerMonitor`.
